Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +16,143 @@ static void Main(string[] args)
+            }
+            return null;
+        }
+
+    }
+
+
+    class Equation{
+
+        public Equation(string operand1,string operand2,string operation,string result)
+        {
+            Operand1 = operand1;
+            Operand2 = operand2;
+            Operation = operation;
+            Result = result;
         }
 
-        public static int FindDigit(string equation)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public string Operand1 { get; private set; }
+        public string Operand2 { get; private set; }
+        public string Operation { get; private set; }
+        public string Result { get; private set; }
+
+        public int CorrectEquationAndReturnMissingDigit()
+        {
+            EquationParts needCorrection = GetEquationPartToBeCorrected();
+            string expectedValue;
+            bool validValue,validEquation;
+            switch(needCorrection)
+            {
+                case EquationParts.Operand1:
+                    expectedValue = Convert.ToString( int.Parse(Result) / int.Parse(Operand2) );
+                    validValue = IsValidValue(expectedValue,Operand1);
+                    validEquation = IsValidEquation(expectedValue,Operand2,Result);
+
+                    if(validValue && validEquation)
+                        return GetIndex(expectedValue,Operand1);
+                    else
+                        return -1;
+
+                case EquationParts.Operand2:
+                    expectedValue = Convert.ToString(int.Parse(Result) / int.Parse(Operand1));
+                    validValue = IsValidValue(expectedValue,Operand2);
+                    validEquation = IsValidEquation(Operand1,expectedValue,Result);
+
+                    if(validValue && validEquation)
+                        return GetIndex(expectedValue,Operand2);
+                    else
+                        return -1;
+
+                case EquationParts.Result:
+                    expectedValue = Convert.ToString(int.Parse(Operand1) * int.Parse(Operand2));
+                    validValue = IsValidValue(expectedValue,Result);
+                    validEquation = IsValidEquation(Operand1,Operand2,expectedValue);
+
+                    if(validValue && validEquation)
+                        return GetIndex(expectedValue,Result);
+                    else
+                        return -1;
+
+                default: 
+                    return -1;
+
+            }
+        }
+
+        private bool IsValidValue(string expectedValue,string givenValue)
+        {
+            return expectedValue.Length==givenValue.Length;
+        }
+
+        private bool IsValidEquation(string operand1,string operand2,string operand3)
+        {
+            return int.Parse(operand1)*int.Parse(operand2) == int.Parse(operand3);
+            
+        }
+
+        private int GetIndex(string str1, string str2){

```

Comment:
Still str1, str2????
'actual' and 'expected' sounds much better.

Null & empty checks are missing on parameters